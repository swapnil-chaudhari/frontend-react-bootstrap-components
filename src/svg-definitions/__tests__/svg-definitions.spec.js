import React from 'react';
import renderShallow from 'render-shallow';
import SVGDefinitions from '../svg-definitions';

describe('<SVGDefinitions>', () => {
    describe('when it renders', () => {
        let component;

        beforeAll(() => {
            component = renderShallow(<SVGDefinitions />).output;
        });

        test('wraps definitions in an <svg> tag', () => {
            expect(component.type).toEqual('svg');
        });

        test('sets the display to none', () => {
            expect(component.props.style).toEqual({
                display: 'block',
                width: 0,
                height: 0
            });
        });

        test('sets the xml namespace to the correct url', () => {
            expect(component.props.xmlns).toEqual('http://www.w3.org/2000/svg');
        });

        describe('svgs', () => {
            test('renders all of the defined svgs', () => {
                expect(component).toMatchSnapshot();
            });
        });
    });
});
