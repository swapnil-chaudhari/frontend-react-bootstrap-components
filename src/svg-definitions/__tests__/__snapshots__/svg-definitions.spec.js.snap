// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SVGDefinitions> when it renders svgs renders all of the defined svgs 1`] = `
<svg
  style={
    Object {
      "display": "block",
      "height": 0,
      "width": 0,
    }
  }
  xmlns="http://www.w3.org/2000/svg"
>
  <SVGAan />
  <SVGAdvertise />
  <SVGArtworkEmpty />
  <SVGAttention />
  <SVGAudioDisabled />
  <SVGAudioEnabled />
  <SVGAudioPlayerMute />
  <SVGAudioPlayerNext />
  <SVGAudioPlayerNextLite />
  <SVGAudioPlayerPause />
  <SVGAudioPlayerPauseLite />
  <SVGAudioPlayerPlay />
  <SVGAudioPlayerPrevious />
  <SVGAudioPlayerPreviousLite />
  <SVGAudioPlayerVolume />
  <SVGCalendar />
  <SVGChevron />
  <SVGCleanVersion />
  <SVGCloseButton />
  <SVGCloseRound />
  <SVGCompleteCheckmark />
  <SVGDigital />
  <SVGDigitalAudio />
  <SVGDownDirectionArrow />
  <SVGDownloadBox />
  <SVGDownloadSymbol />
  <SVGEdit />
  <SVGError404 />
  <SVGExplicit />
  <SVGExternalLink />
  <SVGFacebook />
  <SVGFilledChevronDown />
  <SVGFilledChevronUp />
  <SVGFilm />
  <SVGGear />
  <SVGGoToCatalog />
  <SVGHistory />
  <SVGInfo />
  <SVGLeftPaginationArrow />
  <SVGLoadingArrows />
  <SVGLoadingRing />
  <SVGLogo />
  <SVGLogoAndName />
  <SVGMan />
  <SVGNewProduct />
  <SVGPdf />
  <SVGPhysical />
  <SVGPhysicalAudio />
  <SVGPlay />
  <SVGPlus />
  <SVGRefresh />
  <SVGRightDirectionArrow />
  <SVGRightPaginationArrow />
  <SVGRoyaltyShareLogo />
  <SVGSearch />
  <SVGServerError />
  <SVGSpotify />
  <SVGStatusComplete />
  <SVGStatusInProgress />
  <SVGStatusRed />
  <SVGStatusSubmitted />
  <SVGStop />
  <SVGSuccessCheckmark />
  <SVGThreeDots />
  <SVGThreeRightDirectedArrows />
  <SVGTracksCompleteCheckmark />
  <SVGTracksEmpty />
  <SVGTransparentLogo />
  <SVGTrashCan />
  <SVGTwitter />
  <SVGTripleCheckmark />
  <SVGUpAndDownArrows />
  <SVGUpAndDownCarets />
  <SVGUpDirectionArrow />
  <SVGUpload />
  <SVGViewMoreNextArrow />
  <SVGWoman />
  <SVGXls />
  <SVGYoutube />
  <SVGCamera />
  <SVGPlaylistPlaceholder />
  <SVGUploadArrow />
  <SVGImportArrow />
  <SVGVideo />
  <SVGVideoError />
</svg>
`;
